packages/loader/container-loader/src/container.ts

```diff
@@ -750,9 +761,10 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         assert(this.attachState === AttachState.Attached, 0x0d1 /* "Container should be attached before close" */);
         assert(this.resolvedUrl !== undefined && this.resolvedUrl.type === "fluid",
             0x0d2 /* "resolved url should be valid Fluid url" */);
-        const pendingState: IPendingLocalState = {
+        const pendingState: IPendingContainerState = {
             pendingRuntimeState: this.context.getPendingLocalState(),
             url: this.resolvedUrl.url,
+            clientId: this.clientId,
```

Comment:
The code in ConnectionStateHandler assumes that this clientId is present in quorum. But I do not think it's the case - client may have disconnected and already processed leave op for self when serialization happens. We keep old clientId until we get new one, which happens when we both processed leave op and either join op or connected as "read"